unwind/src/libunwind_shim.rs

```diff
@@ -120,19 +124,28 @@ pub unsafe extern "C" fn _Unwind_FindEnclosingFunction(pc: *mut c_void) -> *mut
 #[no_mangle]
 pub unsafe extern "C" fn _Unwind_RaiseException(exception: *mut _Unwind_Exception) -> _Unwind_Reason_Code {
     (*exception).private_contptr = None;
-    DwarfUnwinder::default().trace(|frames| unwind_tracer(frames, exception));
+    ::glue::registers(|registers| {
+        if let Some(registers) = unwind_tracer(registers, exception) {
+            ::glue::land(&registers);
+        }
```

Comment:
else abort; get rid of the unreachable?

For Rust it should really be unreachable but `_Unwind_RaiseException` is actually supposed to return `_URC_END_OF_STACK` in this case (which would hit [this abort](https://github.com/rust-lang/rust/blob/79d8a0fcefa5134db2a94739b1d18daa01fc6e9f/src/libstd/panicking.rs#L529) in libstd). The only problem is that because we're lazy and skip phase 1 we end up with a half-unwound stack so aborting on our side is probably more correct for now. Either way, the one thing we don't want to happen is another round of panicking.

Reply:
Do you know how to do an abort? Is `std::process::abort` the only option?

Reply:
Also, I was wondering if we can solve our panic recursion by compiling this crate with panic=abort. Is that possible on a per-crate basis?

Reply:
>Also, I was wondering if we can solve our panic recursion by compiling this crate with panic=abort. Is that possible on a per-crate basis?

I don't think it is, sadly. Cargo's `unwind = "abort"` is under `[profile.x]` and thus specific per output binary (so entirely irrelevant for library crates).

>Do you know how to do an abort? Is `std::process::abort` the only option?

If you're looking for something outside of libstd there's still `core::intrinsics::abort` (unstable of course). Finally you could always just FFI to libc's `abort()` but I don't see much of a point in doing that.

Reply:
Going to leave this for now. This PR didn't change the behaviour here, and I think it's better to address panic/abort handling in another PR.